jerboa/finetune.py

```diff
@@ -323,29 +323,32 @@ def generate_and_tokenize_prompt(data_point):
 
     if is_master_process:
         lora_dir = f"{output_dir}/lora_adapter"
-
-        if eval_file:
-            results = evaluate(
-                model=model,
-                tokenizer=tokenizer,
-                eval_file=eval_file,
-                eval_limit=eval_limit,
-            )
-
-            if use_wandb:
-                columns = list(results[0].keys())
-                results_data = [[d[key] for key in columns] for d in results]
-                eval_table = wandb.Table(columns=columns, data=results_data)
-                run.log({"Evaluation": eval_table})
-            else:
-                print(results)
-
-        model.save_pretrained(lora_dir)
-
-        if wandb_log_model and use_wandb:
-            artifact = wandb.Artifact(name='lora_weight', type='model')
-            artifact.add_dir(lora_dir)
-            run.log_artifact(artifact)
+        try:
+            if eval_file:
+                results = evaluate(
+                    model=model,
+                    tokenizer=tokenizer,
+                    eval_file=eval_file,
+                    eval_limit=eval_limit,
+                )
+
+                if use_wandb:
+                    columns = list(results[0].keys())
+                    results_data = [[d[key] for key in columns] for d in results]
+                    eval_table = wandb.Table(columns=columns, data=results_data)
+                    run.log({"Evaluation": eval_table})
+                else:
+                    print(results)
+        except:
+            pass
```

Comment:
I am not a fan at all of excepting any error. This mean even ctrl c will not kill the process.

Tho with python error system it is not easy to know what error could be thrown

Reply:
What type of Exception do you advise?

Would the code not enter the exception on car c, and then you can press ctrl c again to kill it?

Reply:
yeah fair